pg_mon.c

```diff
@@ -641,6 +648,16 @@ static mon_rec * create_or_get_entry(mon_rec temp_entry, int64 queryId)
     return entry;
 }
 
+/* Check if the provided queryId already exists in hash table */
+static bool is_new_entry(int64 queryId){
+
+    bool found = false;
+
+    hash_search(mon_ht, &queryId, HASH_FIND, &found);
```

Comment:
There is a race condition. Many parallel connections have a chance to see a missing element in the hash, but only one of them will insert it while holding an exclusive lock: https://github.com/RafiaSabih/pg_mon/pull/13/files#diff-21e368dadd2ad9016021ced5ec165533f7cda08ce9b684078d40acc3d55e1f81R627-R643

Reply:
Thank you for the review.